Source/WebCore/dom/Document.h

```diff
@@ -853,7 +853,7 @@ class Document
     void disableWebAssembly(const String& errorMessage) final;
     void setTrustedTypesEnforcement(JSC::TrustedTypesEnforcement) final;
 
-    bool requiresTrustedTypes() const { return m_requiresTrustedTypes; }
+    bool requiresTrustedTypes() const { return m_requiresTrustedTypes && !shouldBypassMainWorldContentSecurityPolicy(); }
```

Comment:
Hm... this function is called in pretty hot functions like `Element::setAttribute` and alike.
We should probably schedule perf A/B for speedometer.

Reply:
It might be better to add this check when we set the value rather than access it?